bluesim-rlib/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+//! The Rust library called by bluesim.
+//! If you want to use RProbe in your bluespec project,
+//! please compile this crate into an .a file and then link it to your bluesim executable.
+#![warn(clippy::unwrap_used)]
+use rb_link::{B2RMessage, GetPutMessage, MsgSizeType, MSG_SIZE_BYTES};
+use std::io::{Read, Write};
+use std::os::unix::net::UnixStream;
+use std::sync::OnceLock;
+
+static mut STREAM: OnceLock<UnixStream> = OnceLock::new();
+
+/// # Safety
+/// This function should not be called by Rust code.
+/// Get data from your rust program.
+/// called by RProbe::get_data()
+#[no_mangle]
+pub unsafe extern "C" fn get(res_ptr: *mut u8, id: u32, _cycles: u32, size: u32) {
+    // println!("send get");
+    // check the ptr is not null
+    if res_ptr.is_null() {
+        panic!("res_ptr is a null pointer!");
+    }
+    if _cycles == u32::MAX {
+        panic!("cycles over flow!");
+    }
+    let mut stream = STREAM.get_or_init(|| {
+        UnixStream::connect(String::from("/tmp/b2rr2b")).expect("Failed to connect to socket")
```

Comment:
to speed up test, it's common to run multi test process at the same time.  A hardcoded local unix-stream will not work when there is multi process running different tests.

Reply:
fixed